src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -5638,4 +5653,37 @@ private Set<String> getGeneratedFeatures() {
                 generatedFeaturesFile, null, null);
         return genFeatSet;
     }
+
+    /**
+     * Create metadata when running devc mode and containers
+     * Language server then uses metadata file to connect
+     * 
+     * @throws XMLStreamException
+     * @throws IOException
+     * @throws FactoryConfigurationError
+     */
+    public void writeDevcMetadata() {
+        File metaFile = new File(serverDirectory, "liberty-dev-metadata.xml");
+        try {
+            XMLStreamWriter metadataWriter = XMLOutputFactory.newInstance().createXMLStreamWriter(new FileWriter(metaFile)) ;
+            metadataWriter.writeStartDocument();
+            metadataWriter.writeStartElement("devModeMetaData");
```

Comment:
This file only exists for `devc` mode right? So maybe change this element to `devcModeMetaData`.